src/transformers/generation_tf_utils.py

```diff
@@ -1596,11 +1650,9 @@ def _prepare_encoder_decoder_kwargs_for_generation(self, input_ids: tf.Tensor, m
         }
 
         # vision models don't use `attention_mask`.
-        signature = dict(inspect.signature(encoder.call).parameters)
-        if "attention_mask" not in signature:
-            encoder_kwargs.pop("attention_mask")
```

Comment:
Did you test the script with VisionEncoderDecoder and other vision models that use generate?

Reply:
Only VisionEncoderDecoder. Can run on others -- any other vision models with strong generation tests?